### PR TITLE
Iteration 1 craig

### PR DESCRIPTION
**Pull request by nullreferrer**
Completed refactoring of the ReturnBookControl and ReturnBookUI classes.

### CODE REVIEW

File: src/ReturnBookUI.java [hunk 1]
```diff
@@ -1,87 +1,81 @@
 import java.util.Scanner;
 
-
 public class ReturnBookUI {
 
-	public static enum UI_STATE { INITIALISED, READY, INSPECTING, COMPLETED };
+	public enum UIState { INITIALISED, READY, INSPECTING, COMPLETED };
```
Comment (HenleyCH) [inline src/ReturnBookUI.java:5]: Removing redundant static modifiers is not established in our code guidelines.
Discuss in meeting.

File: src/ReturnBookUI.java [hunk 2]
```diff
@@ -1,87 +1,81 @@
 import java.util.Scanner;
 
-
```
Comment (HenleyCH) [inline src/ReturnBookUI.java:3]: Removing empty line 'whitespace' is not established in the guidelines.
Discuss in meeting.

File: src/ReturnBookUI.java [hunk 3]
```diff
@@ -1,87 +1,81 @@
...
-					CoNtRoL.Scanning_Complete();
+				String bookScan = askQuestion("Scan Book (<enter> completes): ");
+				if (bookScan.length() == 0) {
+					this.control.completeScanning();
 				}
 				else {
 					try {
-						int Book_Id = Integer.valueOf(Book_STR).intValue();
-						CoNtRoL.Book_scanned(Book_Id);
+						int bookId = Integer.valueOf(bookScan).intValue();
```
Comment (HenleyCH) [inline src/ReturnBookUI.java:35]: Need to make decision on 'ID' vs 'Id' or 'id'.
Discuss in meeting.

File: src/ReturnBookUI.java (hunk 1 above)
Comment (nullreferrer) [reply src/ReturnBookUI.java:5]: As discussed, adding to guidelines that redundant static modifiers on enums will be removed, as enums are inherently static.

File: src/ReturnBookUI.java (hunk 2 above)
Comment (nullreferrer) [reply src/ReturnBookUI.java:3]: This isn't an established guideline, it was done in an attempt to better read the code during the refactor.

File: src/ReturnBookUI.java (hunk 3 above)
Comment (nullreferrer) [reply src/ReturnBookUI.java:35]: As discussed "Id" will be the standard as it more closely follows the style guidelines.

File: src/ReturnBookControl.java [hunk 4]
```diff
@@ -1,77 +1,87 @@
...
+
+		double overdueFine = 0.0;
+
+		if (currentLoan.OVer_Due()) {
+			overdueFine = library.CalculateOverDueFine(currentLoan);
+		}
+
+		this.returnBookUI.display("Inspecting");
+		this.returnBookUI.display(currentBook.toString());
+		this.returnBookUI.display(currentLoan.toString());
```
Comment (HenleyCH) [inline src/ReturnBookControl.java:57]: Rule 09 - The argument list of any method calls MUST NOT contain any other method calls.

File: src/ReturnBookControl.java [hunk 5]
```diff
@@ -1,77 +1,87 @@
...
 		
-		if (CurrENT_loan.OVer_Due()) {
-			Ui.display(String.format("\nOverdue fine : $%.2f", Over_Due_Fine));
+		if (this.currentLoan.OVer_Due()) {
+			this.returnBookUI.display(String.format("\nOverdue fine : $%.2f", overdueFine));
```
Comment (HenleyCH) [inline src/ReturnBookControl.java:60]: Rule 09 - The argument list of any method calls MUST NOT contain any other method calls.

File: src/ReturnBookControl.java (hunk 4 above)
Comment (nullreferrer) [reply src/ReturnBookControl.java:57]: Mmmm. Yes you're right here. I missed that, assuming the toString() was a native extension method.

**Review by HenleyCH — CHANGES_REQUESTED** [review]
Method calls inside other methods needs to be addressed in ReturnBookControl.java

**Comment by HenleyCH** [comment]
Please refactor the commented code to meet the code guidelines.

**Pull request closed without merging**